### ISSUE TITLE
Does not download BSE Bhavcopy in Windows 10

### ISSUE THREAD

**Issue by giteshs999** [hemenkapadia/getbhavcopy#104]
![image](https://user-images.githubusercontent.com/21261127/69011623-90f3e680-0992-11ea-8139-485d5786b6b7.png)

It does not do anything and just stays at the message downloading Index Data for Sensex.

**Comment by hemenkapadia** [comment]
@giteshs999, do you still have the issue? Are you able to access BSE using the browser when it seems stuck ?

**Comment by giteshs999** [comment]
Yes, issue is still there. I'm able to download the bhavcopy from BSE website. That is what I'm doing for a while.
If you see, it seems to be stuck while downloading the index data (sensex).

**Comment by neilazavedo** [comment]
Hi Hemen, I too face this same issue .. These days download is only available after 12pm uptil about 8.30pm I'm able to download the NSE Data but BSE freezes for me as well... just like the snapshot uploaded here by Gitesh. Thanks ..

**Comment by neilazavedo** [comment]
Hi Hemen, Please let us know if we have to make some changes to windows 10 settings etc. Thank you.

**Comment by giteshs999** [comment]
Now it has stopped downloading NSE data also. It seems to be stuck at the indexes and keeps giving errors.

![image](https://user-images.githubusercontent.com/21261127/71456082-c673c700-27bd-11ea-9242-b245993c82f3.png)

**Comment by neilazavedo** [comment]
I downloaded this yesterday.. if this helps .. NSE cash 
[2019-12-24-NSE-EQ.txt](https://github.com/hemenkapadia/getbhavcopy/files/4001270/2019-12-24-NSE-EQ.txt)

**Comment by giteshs999** [comment]
Yes, it does work. I guess it is timing issue. So need to download between the time you specified in the earlier message - 12pm up til about 8.30pm. But I can directly go to the NSE website anytime and download.

**Comment by neilazavedo** [comment]
Hi Hemen I am facing the BSE issue in the new version 3.0.1a as well ... NSE is working fine for me. Thank you ...

**Comment by hemenkapadia** [comment]
I am not sure what is the issue with BSE for you. I just verified that 3.0.1a is downloading data for NSE and BSE without issues and you are the only one that is reporting this issue.

Are you sure there is nothing else blocking the BSE request from your machine? Is there a proxy involved ?

**Comment by neilazavedo** [comment]
nope I do not use any proxy ...

**Comment by neilazavedo** [comment]
I have updated the screenshot and log file on the other thread ... https://github.com/hemenkapadia/getbhavcopy/issues/123

**Comment by hemenkapadia** [comment]
See response on https://github.com/hemenkapadia/getbhavcopy/issues/125#issuecomment-570143440 

For me NSE and BSE download works without any issue on 3.0.1a. Unless I can reporiduce the issue there is not much to fix.

**Comment by neilazavedo** [comment]
Yes I agree.. it's something with win10 maybe or the latest .net. ..Absolutely no idea .. ok is there any specific BSE URL I can ping to check if it works from the command prompt etc

**Comment by hemenkapadia** [comment]
I am on Windows 10 and develop GBC on it so I don't think it is Windows 10 specific. 

You can enable Debug mode in Getbhavcopy and share the log. Additionally, the log file has the BSE URLs it is trying to connect to. You can try opening them in your browser. Also check if your browser has any proxy configured.

1. Browse to the directory where you have getbhavcopy.exe file.
2. Delete the log folder existing in that directory.
3. Start the command prompt , Start >> run >> cmd
4. Change to the directory where you have getbhavcopy.exe
5. In that directory start getbhavcopy as 'getbhavcopy.exe debugon'
6. Try to download the data as you do normally.
7. You will see the log directory will be recreated, with a log file inside it.
8. Share the contents of the log file on the github issue.

**Comment by neilazavedo** [comment]
Ok Hemen.. Thanks for the update ... I have upgraded to 3.0.2a .. but face the same issue . I have followed your steps above and the following is the feedback ...

![bse-20200107](https://user-images.githubusercontent.com/58835329/71905160-9816c400-318d-11ea-83fb-41c3c7d7a976.jpg)

Settings : 
![bse-20200107-settings](https://user-images.githubusercontent.com/58835329/71905222-b7155600-318d-11ea-9472-ad394a343deb.jpg)

Log file : 
[log.txt](https://github.com/hemenkapadia/getbhavcopy/files/4031171/log.txt)

Browser : 
![bse-20200107-browser](https://user-images.githubusercontent.com/58835329/71905732-c1841f80-318e-11ea-8b94-42d7763af424.jpg)

![bse-20200107-browser1](https://user-images.githubusercontent.com/58835329/71905803-e5476580-318e-11ea-9bef-a931991b4524.jpg)

**Comment by bubbu64** [comment]
Hi Hemen
Shailendra here, I have been using ur getbhav downloader for last 7-8 yrs without a break. It works flawlessly most of the time however this time referring to the above issue the downloader works great for the BSE on a win 7 OS but not on a win 10 OS. Just downloaded BSE data right from 2000-2020 on my win7 loaded older laptop. But not able to download on my other win10 loaded laptop. On the win 10 laptop NSE/NSE FUTURES work great though. This is FYI. Hope this helps in resolving the issue. 
Regards
Shailendra

**Comment by neilazavedo** [comment]
Maybe you are right bubbu64 ... as mentioned here https://github.com/hemenkapadia/getbhavcopy/issues/142 Win 10 Home has an issue ..

**Comment by hemenkapadia** [comment]
> 
> 
> Maybe you are right bubbu64 ... as mentioned here #142 Win 10 Home has an issue ..

I don't think so. There is some other issue on your machines. I develop and test GBC on Windows 10 Home. This is my windows laptop OS version.

![image](https://user-images.githubusercontent.com/9943123/72239403-d1886e80-3595-11ea-938f-0006cbceadc2.png)

And BSE works flawlessly for me. There is something else blocking.

**Comment by hemenkapadia** [comment]
@neilazavedo - 

what does the below link do in browser 

https://api.bseindia.com/BseIndiaAPI/api/IndexArchDaily/w?period=D&index=SENSEX&fmdt=09/01/2020&todt=09/01/2020

**Comment by hemenkapadia** [comment]
Here is the output of the curl command on my windows. Please take note of the full curl command, especially quoting the URL.

Your Windows 10 may have curl installed by default. If not, this page has good information - https://develop.zendesk.com/hc/en-us/articles/360001068567-Installing-and-using-cURL


```
D:\Hemen\Workspace\Getbhavcopy-Code\Getbhavcopy\Getbhavcopy_v2\Getbhavcopy_v2\bin\Release\Protected>curl -vv "https://api.bseindia.com/BseIndiaAPI/api/IndexArchDaily/w?period=D&index=SENSEX&fmdt=09/01/2020&todt=09/01/2020"
*   Trying 2600:1406:1400:581::f5...
* TCP_NODELAY set
*   Trying 104.80.207.76...
* TCP_NODELAY set
* Connected to api.bseindia.com (104.80.207.76) port 443 (#0)
* schannel: SSL/TLS connection with api.bseindia.com port 443 (step 1/3)
* schannel: checking server certificate revocation
* schannel: sending initial handshake data: sending 181 bytes...
* schannel: sent initial handshake data: sent 181 bytes
* schannel: SSL/TLS connection with api.bseindia.com port 443 (step 2/3)
* schannel: failed to receive handshake, need more data
* schannel: SSL/TLS connection with api.bseindia.com port 443 (step 2/3)
* schannel: encrypted data got 3806
* schannel: encrypted data buffer: offset 3806 length 4096
* schannel: sending next handshake data: sending 126 bytes...
* schannel: SSL/TLS connection with api.bseindia.com port 443 (step 2/3)
* schannel: encrypted data got 242
* schannel: encrypted data buffer: offset 242 length 4096
* schannel: SSL/TLS handshake complete
* schannel: SSL/TLS connection with api.bseindia.com port 443 (step 3/3)
* schannel: stored credential handle in session cache
> GET /BseIndiaAPI/api/IndexArchDaily/w?period=D&index=SENSEX&fmdt=09/01/2020&todt=09/01/2020 HTTP/1.1
> Host: api.bseindia.com
> User-Agent: curl/7.55.1
> Accept: */*
>
* schannel: client wants to read 102400 bytes
* schannel: encdata_buffer resized 103424
* schannel: encrypted data buffer: offset 0 length 103424
* schannel: encrypted data got 782
* schannel: encrypted data buffer: offset 782 length 103424
* schannel: decrypted data length: 753
* schannel: decrypted data added: 753
* schannel: decrypted data cached: offset 753 length 102400
* schannel: encrypted data buffer: offset 0 length 103424
* schannel: decrypted data buffer: offset 753 length 102400
* schannel: schannel_recv cleanup
* schannel: decrypted data returned 753
* schannel: decrypted data buffer: offset 0 length 102400
< HTTP/1.1 200 OK
< Pragma: no-cache
< Content-Type: application/json; charset=utf-8
< Server: BSELTD
< X-AspNet-Version: 4.0.30319
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET,PUT,POST,DELETE,OPTIONS
< Cache-Control: no-cache
< Expires: Mon, 13 Jan 2020 08:04:22 GMT
< Date: Mon, 13 Jan 2020 08:04:22 GMT
< Content-Length: 282
< Connection: keep-alive
< Server-Timing: cdn-cache; desc=REVALIDATE
< Server-Timing: edge; dur=29
< Server-Timing: origin; dur=494
<
{"Table":[{"Day":9,"Month":"January","year":"2020","Turnover":"543.94","tdate":"2020-01-09T00:00:00","I_name":"SENSEX ","I_open":41216.67,"I_high":41482.12,"I_low":41175.72,"I_close":41452.35,"I_pe":26.22,"I_pb":3.24,"I_yl":1.02,"Turnover_1":"543.94","TOTAL_SHARES_TRADED":"0.78"}]}* Connection #0 to host api.bseindia.com left intact

D:\Hemen\Workspace\Getbhavcopy-Code\Getbhavcopy\Getbhavcopy_v2\Getbhavcopy_v2\bin\Release\Protected>
```

**Comment by neilazavedo** [comment]
Hi Hemen.. I get the following ..
![browser-tab-json](https://user-images.githubusercontent.com/58835329/72244828-1acbd300-3615-11ea-8d28-03e833e9a7d7.jpg)
![browser-tab-rawdata](https://user-images.githubusercontent.com/58835329/72244846-20c1b400-3615-11ea-83fb-e656c3131767.jpg)
![browser-tab-headers](https://user-images.githubusercontent.com/58835329/72244866-2cad7600-3615-11ea-8942-946767e2acee.jpg)

**Comment by bubbu64** [comment]
Hemen 
My  win10 OS version is the same as urs 1903 just checked , still BSE doesnt get downloaded, the other two get done. 
Regards
Shailendra

**Comment by bubbu64** [comment]
![Untitled](https://user-images.githubusercontent.com/29131454/72245233-16ec8080-3616-11ea-8c52-439956d4551b.png)

**Comment by bubbu64** [comment]
![Untitled](https://user-images.githubusercontent.com/29131454/72245305-40a5a780-3616-11ea-82f3-681700ba8e00.png)

**Comment by rkarthik1991** [comment]
Hi,
I went through the comments and found that there is unsolved issue with bse data downloading. I downloaded the latest version of getbhavcopy and noticed that the issue is persistent in Win10. Not sure about other OS. Tried every possible option but no use. I'm not using any proxy too.
